java/code/src/com/redhat/rhn/frontend/action/systems/audit/ScapSetupAction.java

```diff
@@ -15,57 +15,97 @@
 
 import com.redhat.rhn.domain.rhnpackage.PackageFactory;
-import com.redhat.rhn.domain.server.InstalledPackage;
 import com.redhat.rhn.domain.server.MinionServer;
 import com.redhat.rhn.domain.server.Server;
+import com.redhat.rhn.domain.server.ServerConstants;
 import com.redhat.rhn.domain.user.User;
 import com.redhat.rhn.frontend.action.systems.sdc.SdcHelper;
 import com.redhat.rhn.frontend.struts.RequestContext;
 import com.redhat.rhn.frontend.struts.RhnAction;
 import com.redhat.rhn.manager.audit.ScapManager;
 
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 /**
  * ScapSetupAction
  */
 
 public abstract class ScapSetupAction extends RhnAction {
-    private static final String SCAP_ENABLED = "scapEnabled";
-    private static final String REQUIRED_PKG = "requiredPackage";
+    private static final Logger LOGGER = LogManager.getLogger(ScapSetupAction.class);
+
+    public static final String SCAP_ENABLED = "scapEnabled";
+    public static final String REQUIRED_PKG = "requiredPackage";
+
     private static final String OPENSCAP_SUSE_PKG = "openscap-utils";
     private static final String OPENSCAP_REDHAT_PKG = "openscap-scanner";
-    private static final String OPENSCAP_DEBIAN_PKG = "libopenscap8";
+    private static final String OPENSCAP_DEBIAN_PKG = "openscap-utils";
+    private static final String OPENSCAP_DEBIAN_LEGACY_PKG = "libopenscap8";
 
     protected void setupScapEnablementInfo(RequestContext context) {
         Server server = context.lookupAndBindServer();
         User user = context.getCurrentUser();
-        boolean enabled = false;
-        String requiredPkg = "";
+
+        boolean enabled;
+        String requiredPackage;
+
         if (server.asMinionServer().isPresent()) {
             MinionServer minion = server.asMinionServer().get();
             switch (minion.getOsFamily()) {
                 case "Suse":
-                    requiredPkg = OPENSCAP_SUSE_PKG;
+                    requiredPackage = OPENSCAP_SUSE_PKG;
                     break;
 
                 case "Debian":
-                    requiredPkg = OPENSCAP_DEBIAN_PKG;
+                    requiredPackage = getPackageForDebianFamily(minion.getOs(), minion.getRelease());
                     break;
 
                 default:
-                    requiredPkg = OPENSCAP_REDHAT_PKG;
-            }
-            InstalledPackage installedPkg =
-                    PackageFactory.lookupByNameAndServer(requiredPkg, server);
-            if (installedPkg != null) {
-                enabled = true;
+                    requiredPackage = OPENSCAP_REDHAT_PKG;
             }
+
+            // Verify the packages is installed
+            enabled = PackageFactory.lookupByNameAndServer(requiredPackage, server) != null;
         }
         else {
             enabled = ScapManager.isScapEnabled(server, user);
+            requiredPackage = "";
         }
```

Comment:
@mcalmer Should we drop this `else` or throw an `IllegalStateException`? We should always have minion in 5.0 so `server.asMinionServer().isPresent()` should always be `true`, right?

Reply:
We have the case that customers might migrate to 5.0, but still have traditional clients. 
This is not supported, but we might have the case. Maybe we keep it for a short while.

Kevin also started an issue with some ideas what we want to do with MinionServer class and DB table at all.
I think some cleanups in this area will come in the near future anyway